src/time.ts

```diff
@@ -1,27 +1,65 @@
 /**
- * A millisecond.
+ * Common duration constants, in milliseconds.
  */
-export const MILLISECOND = 1;
+export enum Duration {
+  /**
+   * A millisecond.
+   */
+  Millisecond = 1,
 
-/**
- * A second, in milliseconds.
- */
-export const SECOND = 1000; // MILLISECOND * 1000
+  /**
+   * A second, in milliseconds.
+   */
+  Second = 1000, // Millisecond * 1000
 
-/**
- * A minute, in milliseconds.
- */
-export const MINUTE = 60_000; // SECOND * 60
+  /**
+   * A minute, in milliseconds.
+   */
+  Minute = 60_000, // Second * 60
 
-/**
- * An hour, in milliseconds.
- */
-export const HOUR = 3_600_000; // MINUTE * 60
+  /**
+   * An hour, in milliseconds.
+   */
+  Hour = 3_600_000, // Minute * 60
+
+  /**
+   * A day, in milliseconds.
+   */
+  Day = 86_400_000, // Hour * 24
+
+  /**
+   * A week, in milliseconds.
+   */
+  Week = 604_800_000, // Day * 7
+
+  /**
+   * A year, in milliseconds.
+   */
+  Year = 31_536_000_000, // Day * 365
+}
+
+const isNonnegativeInteger = (number: number) =>
+  Number.isInteger(number) && number >= 0;
+
+const assertIsNonnegativeInteger = (number: number, name: string) => {
+  if (!isNonnegativeInteger(number)) {
+    throw new Error(
+      `"${name}" must be a non-negative integer. Received: "${number}".`,
+    );
+  }
+};
```

Comment:
This is a bit easier to read:
```suggestion
const isNonNegativeInteger = (number: number) =>
  Number.isInteger(number) && number >= 0;

const assertIsNonNegativeInteger = (number: number, name: string) => {
  if (!isNonnegativeInteger(number)) {
    throw new Error(
      `"${name}" must be a non-negative integer. Received: "${number}".`,
    );
  }
};
```